src/common/hwinfo.cpp

```diff
@@ -0,0 +1,112 @@
+///////////////////////////////////////////////////////////////////////////////
+// Name:        wx/hwinfo.h
+// Purpose:     declaration of the wxSystemHardwareInfo class
+// Author:      Blake Madden
+// Created:     2024-07-09
+// Copyright:   (c) 2024 Blake Madden
+// Licence:     wxWindows licence
+///////////////////////////////////////////////////////////////////////////////
+
+// ============================================================================
+// declarations
+// ============================================================================
+
+// ----------------------------------------------------------------------------
+// headers
+// ----------------------------------------------------------------------------
+
+// for compilers that support precompilation, includes "wx.h".
+#include "wx/wxprec.h"
+
+#include "hwinfo.h"
+
+#ifndef WX_PRECOMP
+    #include "wx/thread.h"
+#endif //WX_PRECOMP
+
+#ifdef __WXMSW__
+    #include <windows.h>
+    #include <sysinfoapi.h>
+#elif defined(__APPLE__)
+    #include <sys/sysctl.h>
+#elif defined(__UNIX__)
+    #include <sys/resource.h>
+    #include <sys/sysinfo.h>
+#endif
```

Comment:
I think it would be better to have platform-specific files rather than put everything in a single file here.